extensions/amp-analytics/0.1/events.js

```diff
@@ -138,13 +138,37 @@ export class CustomEventTracker extends EventTracker {
         });
       }, 1);
     }
+    let selector = config['selector'];
+    if (!selector) {
+      selector = ':root';
+    }
+    const selectionMethod = config['selectionMethod'] || null;
+
+    let targetReady = null;
+    if (selector == ':root' || selector == ':host') {
+      targetReady = Promise.resolve(
+          this.root.getElement(context, selector, selectionMethod));
+    } else {
+      // Wait for DOM to be fully parsed to avoid false missed searches.
+      targetReady = this.root.ampdoc.whenReady().then(() => {
```

Comment:
I added `if()` above to skip checking for document ready if selector is `:root`. I figure it can help trigger handler earlier in those cases. 
Is it not helping?